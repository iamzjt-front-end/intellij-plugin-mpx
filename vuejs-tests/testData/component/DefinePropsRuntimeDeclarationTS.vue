<script setup lang="ts">
defineProps({
  optional: String,
  optionalUnion: [String, Number],
  required: { type: String, required: true },
  requiredUnion: { type: [String, Number], required: true },
});

<error descr="Unresolved variable or type optional">optional</error>;
</script>

<template>
  {{optional}}
  {{optionalUnion}}
  {{required}}
  {{requiredUnion}}
</template>
