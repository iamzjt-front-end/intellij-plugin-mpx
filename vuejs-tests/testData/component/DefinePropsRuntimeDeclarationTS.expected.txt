{
 class: "VueSourceComponent",
 components: {
  DefinePropsRuntimeDeclarationTS: <recursive value of class VueSourceComponent>,
 },
 props: [
  {
   class: "VueSourceInputProperty",
   name: "optional",
   jsType: "string",
   required: false,
  },
  {
   class: "VueSourceInputProperty",
   name: "optionalUnion",
   jsType: "string | number",
   required: false,
  },
  {
   class: "VueSourceInputProperty",
   name: "required",
   jsType: "string",
   required: true,
  },
  {
   class: "VueSourceInputProperty",
   name: "requiredUnion",
   jsType: "string | number",
   required: true,
  },
 ],
}
